conftest.py

```diff
@@ -127,3 +131,43 @@ def pytest_configure(config):
     # tests, and indeed if we do they fail.
     settings.STATICFILES_STORAGE = (
         'django.contrib.staticfiles.storage.StaticFilesStorage')
+
+class CatalogMocker:
+    """A CatalogMocker.
+    A instance of CatalogMocker is a (reusable) context manager.
+    """
+    def __init__(self):
+        self.patcher = mock.patch('ideascube.serveradmin.catalog.Catalog',
+                                  spec=True)
+        self.packages = []
+        self.mocked = None
+
+    def create_package(self, id, **kwargs):
+        pseudo_package = Package(id, kwargs)
+        self.packages.append(pseudo_package)
+
+    def list_installed(self, ids_list):
+        if '*' in ids_list:
+            return self.packages
+
+        return [p for p in self.packages if p.id in ids_list]
+
+    def __enter__(self):
+        if self.mocked is not None:
+            raise RunTimeError(
+                "CatalogMocker is not a reentrant context manager")
```

Comment:
Are we, high-level Python developers, all familiar with the concept of reentrancy? Or should that message just state explicitly that you can't use the same `CatalogMocker` instance as context managers without the first one having exited first?

In general, I tend to prefer messages to explain how to fix the issue when it is a programmer error, rather than just stating the issue.